src/containers/booklist.js

```diff
@@ -2,52 +2,68 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import Book from '../components/book';
-import { removeBook } from '../actions';
+import { removeBook, changeFilter } from '../actions';
+import CategoryFilter from '../components/categoryFilter';
 
-const BookList = ({ books, removeBook }) => {
+const BookList = ({
+  books, category, removeBook, changeFilter,
+}) => {
   const handleRemoveBook = book => {
     removeBook(book.id);
   };
 
+  const handleFilterChange = e => {
+    const filter = e.target.value;
+    changeFilter(filter);
+  };
+
+  const author = 'Hillary';
+
+  const filtered = category === 'All' ? books : books.filter(book => book.category === category);
   return (
     <div>
-      <table>
-        <tr>
-          <th>ID</th>
-          <th>Title</th>
-          <th>Category</th>
-          <th>Delete</th>
-        </tr>
-        {books.map(book => (
+      <CategoryFilter
+        filter={category}
+        handleFilterChange={handleFilterChange}
+      />
+      <div className="books-container">
+        {filtered.map(book => (
           <Book
             id={book.id}
             key={book.id}
+            author={author}
             handleRemoveBook={handleRemoveBook}
             // eslint-disable-next-line react/jsx-props-no-spreading
             {...book}
           />
         ))}
-      </table>
+      </div>
     </div>
   );
 };
 
 const mapStateToProps = state => ({
-  books: state.books,
+  books: state.books.books,
+  category: state.books.filter,
 });
 
 BookList.propTypes = {
-  books: PropTypes.objectOf,
+  books: PropTypes.arrayOf(PropTypes.any),
```

Comment:
According to the previous @mosaaleb told you to use `props-types.shape`

Using [PropTypes.shape](https://dev.to/cesareferrari/how-to-specify-the-shape-of-an-object-with-proptypes-3c56) here will allow you to set a type check that is more accurate, then you can remove the inline comment to disable prop-types.